src/oic/utils/authn/client.py

```diff
@@ -310,14 +311,16 @@ def construct(self, cis, request_args=None, http_args=None, **kwargs):
         # audience is the OP endpoint
         # audience = self.cli._endpoint(REQUEST2ENDPOINT[cis.type()])
         # OR OP identifier
-        audience = self.cli.provider_info['issuer']
         algorithm = None
         if kwargs['authn_endpoint'] in ['token', 'refresh']:
             try:
                 algorithm = self.cli.registration_info[
                     'token_endpoint_auth_signing_alg']
             except (KeyError, AttributeError):
                 pass
+            audience = self.cli.provider_info['token_endpoint']
+        else:
```

Comment:
The drop in coverage is caused by this branch, which is never covered.

Reply:
Can easily be remedied